Zend/zend_inheritance.c

```diff
@@ -167,6 +167,20 @@ char *zend_visibility_string(uint32_t fn_flags) /* {{{ */
 }
 /* }}} */
 
+static zend_bool zend_iterable_type_check(zend_arg_info *arg_info) /* {{{ */
+{
+	if (arg_info->type_hint == IS_ITERABLE || arg_info->type_hint == IS_ARRAY) {
+		return 1;
+	}
+	
+	if (arg_info->class_name && zend_string_equals_literal_ci(arg_info->class_name, "Traversable")) {
+		return 1;
+	}
```

Comment:
Allows a degree of co/contravariance in extending/implementing classes. Parameter types of `array` or `Traversable` can be broadened to `iterable` or return types declaring `iterable` can be restricted to either `array` or `Traversable`.